### ISSUE TITLE
Spring Boot 2.4.7 issue when spring.config.import is used in application-{profile}.yaml

### ISSUE THREAD

**Issue by tekener** [awspring/spring-cloud-aws#151]
**Type**: Bug

**Component**:
Parameter Store

**Describe the bug**
After the update from spring boot 2.4.6 to 2.4.7 this kind of import inside a application-{profile}.yaml does not work anymore.
```
spring:
  config:
    import: 'aws-parameterstore:/config/my-service/'
```

Spring does now only call the `resolve` method on the `AwsParamStoreConfigDataLocationResolver` and not the profile specific `resolveProfileSpecific` one.

It i working when you are doing this in the application.yaml because then the `resolveProfileSpecific` method is then called passing in the profiles.
Tried this workaround but then my local profile without aws access also tries to connect to aws and the spring boot startup will fail.
Setting the import to optional does also not help, because the aws call happens anyway.

**Comment by tekener** [comment]
Will provide a PR with a proper implementation of the resolve method when the import is done in the profile specific config file.

**Comment by tekener** [comment]
A workaround for this issue is to set the import via the command line and not inside the profile specific application.properties
`-Dspring.config.import=aws-parameterstore:/config/my-service/`

**Comment by eddumelendez** [comment]
@tekener I think it is related to https://github.com/spring-projects/spring-boot/issues/26960 which has just been released for 2.4.8

**Comment by maciejwalkowiak** [comment]
I confirm, this issue has been fixed in Spring Boot 2.4.8.
